upip/upip_utarfile.py

```diff
@@ -39,7 +39,13 @@ def readinto(self, buf):
         return sz
 
     def skip(self):
-        self.f.read(self.content_len + self.align)
+        sz = self.content_len + self.align
+        if sz:
+            buf = bytearray(16)
+            while sz:
+                s = min(sz, 16)
+                self.f.readinto(buf, s)
+                sz -= s
```

Comment:
I guess you didn't want to rely on f having a seek method.  But would it be worth demanding this?  DecompIO.seek would have other uses.